### PR TITLE
Add 3rd party conf support for telemetry

### LINKED ISSUE #691: (backlog)Add configuration for all 3rd party telemetry interfaces
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:

Add 3rd party conf support for telemetry

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #691 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@leonwanghui @skdwriting @xing-yang @NajmudheenCT please review, thanks.

File: pkg/api/controllers/metrics.go [hunk 1]
```diff
@@ -56,21 +56,20 @@ var BackupExtension string
...
 
-	// TODO Prakash read these from conf and save to these variables
-	ReloadPath = "/-/reload"
+	ReloadPath = CONF.OsdsLet.ConfReloadUrl
 	BackupExtension = ".bak"
 
-	PrometheusConfHome = "/etc/prometheus/"
-	PrometheusUrl = "http://localhost:9090"
-	PrometheusConfFile = "prometheus.yml"
+	PrometheusConfHome = CONF.OsdsLet.PrometheusConfHome
```
[the remark this answers is not in the document]
Comment (rhsakarpos) [reply pkg/api/controllers/metrics.go]: Hi, sorry, did not understand...this is in the controller code, and OsdsLet will have the configuration values, right ?

[the remark this answers is not in the document]
Comment (leonwanghui) [reply pkg/api/controllers/metrics.go]: No, this is in the apiserver code, controller code is located in `pkg/controller/`.

[the remark this answers is not in the document]
Comment (rhsakarpos) [reply pkg/api/controllers/metrics.go]: fixed this, moved the flags across controller and API server

File: pkg/utils/config/config_define.go [hunk 2]
```diff
@@ -27,14 +27,30 @@ type OsdsApiServer struct {
...
 }
 
 type OsdsLet struct {
 	ApiEndpoint       string        `conf:"api_endpoint,localhost:50049"`
 	Daemon            bool          `conf:"daemon,false"`
 	LogFlushFrequency time.Duration `conf:"log_flush_frequency,5s"` // Default value is 5s
-	GrafanaUrl        string        `conf:"grafana_url,http://localhost:3000"`
-	AlertMgrUrl       string        `conf:"alertmgr_url,http://localhost:9093"`
+	// how to push metrics to Prometheus ? options are PushGateway or NodeExporter
+	PrometheusPushMechanism string `conf:"prometheus_push_mechanism,NodeExporter"`
```
Comment (leonwanghui) [inline pkg/utils/config/config_define.go]: I‘m wondering where is this field used in the code?

Comment (rhsakarpos) [reply pkg/utils/config/config_define.go]: This is used in the metrics adapters...to choose whether to send to node_exporter or to the push gateway.
pkg/controller/metrics/adapters/metrics_sender_to_prometheus.go:50

```
if CONF.OsdsLet.PrometheusPushMechanism == "NodeExporter" {
	// do the actual sending work here, by writing to the file of the node_exporter of prometheus
	writeToFile(work)
	log.Info("GetMetricsSenderToPrometheus processed metrics write to node exporter")
} else if CONF.OsdsLet.PrometheusPushMechanism == "PushGateway" {
	// alternatively, we could also push the metrics to the push gateway of prometheus
	sendToPushGateway(work)
	log.Info("GetMetricsSenderToPrometheus processed metrics send to push gateway")
}
```

**Pull request merged**